runtime/doc/lua.txt

```diff
@@ -3039,6 +3039,11 @@ vim.fs.find({names}, {opts})                                   *vim.fs.find()*
                  • {limit}? (`number`, default: `1`) Stop the search after
                    finding this many matches. Use `math.huge` to place no
                    limit on the number of matches.
+                 • {skip}? (`fun(dir: string): boolean`) Do not traverse
```

Comment:
please search for "predicate", "filter", "skip", and any other related concept, in the :help docs to see which is the most common. I'm wondering if positive logic ("match") or negative logic ("skip") is more common and more intuitive.

Reply:
### Skip
    - vim.fs.dir() - mentions both predicate and skip 
    - vim.iter:skip()
    - vim.iter:rskip()
    - search()
    - searchparipos()
    - searchpos()
    - strcharpart()
    - strchars()

### Filter
    - filter()
    - vim.inspect_pos()
    - vim.show_pos()
    - vim.tbl_filter()
    - vim.iter:filter()
    - vim.snippet.active()
    - vim.lsp.buf.code_action() - mentions both filter and predicate
    - vim.lsp.buf.format() - mentions both filter and predicate
    - vim.lsp.buf.rename() - mentions both filter and predicate
    - `filter` is mentions as prefer way to control `enable` functionality in dev-name-common and dev-api-patterns

### Match
    - string:match()
    - string:gmatch()
    - vim.lpeg.Pattern:match()
    - vim.filetype.match()
    - nvim_get_autocmds()
    - vim.regex:match_line()
    - vim.regex:match_str()
    - vim.diagnostic.match()
    - match() and match*() family

### Predicate
    - vim.fs.dir() - mentions both predicate and skip
    - vim.tbl_contains()
    - vim.tbl_filter()
    - vim.iter:all()
    - vim.iter:any()
    - vim.iter:find()
    - vim.iter:rfind()
    - vim.lsp.start() - reuse_client
    - vim.lsp.buf.code_action() - mentions both filter and predicate
    - vim.lsp.buf.format() - mentions both filter and predicate
    - vim.lsp.buf.rename() - mentions both filter and predicate
    - used by TS API

I might had missed some cases but I think this is a general overview of idioms.

it looks like `match` is by far the prefer idiom in the code base since it's used in a lot places to filter/narrow down in various scenarios, although is not so commonly used in as filter mechanism for functions/callbacks, `skip` is actually not that commonly used but is what I picked for `vim.fs.find` since I borrowed the concept from `vim.fs.dir`

Reply:
> I borrowed the concept from `vim.fs.dir`

Oh, I missed that an existing `vim.fs` function already has this concept. In that case this seems fine.